### DISCUSSION TITLE
Doppler and OFDM usage

### DISCUSSION THREAD

**Discussion by Priyanshjain24** [NVlabs/sionna#516] [Q&A]
Hi. I was trying to understand how to use the apply_doppler and the cir_to_ofdm_channel and it otuput meaning. I am using the simple_street_canyon_with_Cars scenario with only two cars, each having a single antenna. I have a two antenna reciever also in the scenario. After calculating the paths using scene.compute_paths I wanted to use the apply_doppler function. I am not sure on how to pass the velocities to it. Currently I am passing a list containing the velocities of the two cars (also the transmitters) to it. After this, when I use the cir_to_ofdm_channel and tf.sqeeze I get the output shape as TensorShape([2, 2, 1024, ] indicate1024]). I do not understand what the first two dimensions are for. I am interested in finding out the channel between the transmitter and a reciever (which I think should be a 2D array). Any help wiuld be appreciated.

**Comment by SebastianCa** [comment]
Hi @Priyanshjain24,
please have a look at the [Ray tracing tutorial](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Introduction.html). In particular, the [Section on CIRs](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Introduction.html#From-Paths-to-Channel-Impulse-Responses) gives a detailed explanation about the tensor dimensions. There is also a [tutorial related to mobility](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Mobility.html) in the Sionna ray tracer available.

**Comment by Priyanshjain24** [comment]
Hi @SebastianCa,
Thank you for the links. I have gone through them and according to the documentation of cir_to_ofdm_channel the first two dimenstions means num_tx=2 and num_tx_ant=2. But I still do not understand why there are two ofdm channel for each transmitter when I have only a single antenna, When I use the ApplyOFDM channel function, it outputs 2 ofdm channels which are identical. This makes sense that due to the fact that I have a single antenna and multiple transmitters. I am doing everything as here [Section on CIRs](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Introduction.html#From-Paths-to-Channel-Impulse-Responses). Can you suggest some way of making link between a single antenna of a reciever and a transmitter?

**Comment by jhoydis** [comment]
Hi @Priyanshjain24,

It seems that you have two transmitters and a single two-antenna receiver. 

The output shape of ``cir_to_ofdm_channel`` is [batch size, num_rx, num_rx_ant, num_tx, num_tx_ant, num_time_steps, fft_size].

In your case, this would be:
[1, 1, 2, 2, 1, 1, num_time_steps, fft_size]

How many time steps did you simulate in ``apply_doppler()``? What is the FFT size (I assume 1024)?

You mentioned that you were applying a squeeze operation to the output which would remove all dimensions equal to one, i.e.,:

[2,2,num_time_steps, fft_size].

If you have only a single time step, this dimension would also be removed. That should explain the behavior you observe. 

The entry [0,0] would describe the channel frequency response from the first transmitter to the first antenna of the receiver. The entry [1, 0] would describe the channel frequency response from the first transmitter to the second antenna of the receiver.

Hope that answers your question.

**Comment by Priyanshjain24** [comment]
Hi. I simulated it for 1024 time steps.

**Reply by jhoydis** [reply]
And which FFT size did you use, just 1?

**Reply by Priyanshjain24** [reply]
1024

**Reply by jhoydis** [reply]
Then the output dimensions must be [2,2,1024,1024] (after doing the squeeze).

**Reply by Priyanshjain24** [reply]
yes. But I don't understand why there are 2*2 channels when there are only two trnasmitters.

**Reply by jhoydis** [reply]
I was trying to explain it above. You have two transmitters with a single antenna and one receiver with 2 antennas. There is a channel between every transmit antenna and every receive antenna, i.e., 2x2.

**Reply by Priyanshjain24** [reply]
Oh Okay. What I want to do is find out the channel between the transmitter and the receiver. Is there any way to link single antenna of the receiver to a particular transmitter? Or is there some other way to do this?

**Reply by jhoydis** [reply]
You have a tensor of shape h= [2,2,1024,1024] that contains the channel frequency responses between all transmitters and receivers across 1024 subcarrier for 1024 subsequent OFDM symbols.
If you want to know the channel from the jth transmitter to the ith receiver antenna, just pick h[i,j].

**Reply by Priyanshjain24** [reply]
Thank you so much. I was able to understand the stuff and also solve my issue. Thanks again
